In [2]:
import pandas as pd
from datetime import datetime

In [3]:
price = 9.99
dev_proc = 0.7

In [4]:
df = pd.read_csv('data_analytics.csv')

In [5]:
# Data type formatting
df['Event Date'] = df['Event Date'].apply(lambda x: datetime.strptime(x, '%Y-%m-%d'))

In [6]:
# Let's group data by unique customer id and count how many times it does conversion action
grouped = df.groupby(by='Subscriber ID', as_index=False)['Event Date'].count()-1
grouped.rename(columns={'Event Date': 'Purchase'}, inplace=True)

In [7]:
# Using formula to calculate LTV
first = dev_proc*price
grouped['LTV'] = grouped['Purchase']*first

In [8]:
print(grouped)

           Subscriber ID  Purchase     LTV
0     223411322456765432         5  34.965
1     223411322456765433         5  34.965
2     223411322456765434         5  34.965
3     223411322456765436         5  34.965
4     223411322456765437         5  34.965
...                  ...       ...     ...
1014  223411322456766799         0   0.000
1015  223411322456766800         0   0.000
1016  223411322456766801         0   0.000
1017  223411322456766804         0   0.000
1018  223411322456766807         0   0.000

[1019 rows x 3 columns]


In [9]:
# Let's get average value
LTV_for_6_weeks = grouped['LTV'].mean()

In [11]:
print(LTV_for_6_weeks)

9.340012757605495


<b> It means that in period of 6 weeks marketing team need to spend not more than 9.34 to attract new  unique
customers